api/api.go

```diff
@@ -200,6 +200,16 @@ type Config struct {
 	TLSConfig *TLSConfig
 
 	Headers http.Header
+
+	// url is populated with the initial parsed address and is not modified in the
+	// case of a unix:// URL, as opposed to Address.
+	url *url.URL
+}
+
+// URL returns a copy of the initial parsed address and is not modified in the
+// case of a `unix://` URL, as opposed to Address.
+func (c *Config) URL() *url.URL {
+	return c.url
 }
```

Comment:
Note to other reviewers: @angrycub and I paired on this for quite a while. The code here in the `api` package handles the `unix://` scheme correctly now without doing a bunch of weird string manipulation. But the `nomad operator api` package has some special case logic around emitting `curl`-alike outputs. Rather than exposing flags or fields special-casing the UDS, we landed on this method, which is potentially more generally useful for callers and lets the `nomad operator api` check the scheme directly.